zarr/storage.py

```diff
@@ -1113,7 +1113,9 @@ def rmdir(self, path=None):
             raise ReadOnlyError()
         store_path = self.dir_path(path)
         if self.fs.isdir(store_path):
-            self.fs.rm(store_path, recursive=True)
+            # sometimes doesn't delete on first try with S3
+            while self.fs.exists(store_path):
+                self.fs.rm(store_path, recursive=True)
```

Comment:
We may want to have exponential backoff with a timeout. But that can be done later in a subsequent PR.

Reply:
This shouldn't be required - did you figure out what it going on? Perhaps just time for the metadata to sync?

Reply:
@martindurant, I tried a `time.sleep` for various times (5, 10, 30 seconds) and that didn't seem to work. This is the only way I could figure out how to get it to pass without changing [test_s3_complex](https://github.com/zarr-developers/zarr-python/blob/06450d038aa4bd5502e1a72eee824c1d5f409494/zarr/tests/test_storage.py#L958-L999). Another way that I could get it to pass was by running :
```python
a = g.create_dataset("data", shape=(8, 8, 8), mode='w',
                             fill_value=-1, chunks=(1, 1, 1), overwrite=True)
a = g.create_dataset("data", shape=(8, 8, 8), mode='w',
                             fill_value=-1, chunks=(1, 1, 1), overwrite=True)
```
in place of the code [here](https://github.com/zarr-developers/zarr-python/blob/06450d038aa4bd5502e1a72eee824c1d5f409494/zarr/tests/test_storage.py#L997) and that also seemed to resolve the problem. For what it's worth, this test does't fail if fsspec is set to `0.8.3`, ony when its set to `0.8.4`, although at `0.8.3`, [test_pickle](https://github.com/zarr-developers/zarr-python/blob/06450d038aa4bd5502e1a72eee824c1d5f409494/zarr/tests/test_core.py#L759-L796) will fail.